### PR TITLE
[New Rule]  AWS IAM Backdoor User Key Creation

### LINKED ISSUE #1128: [New Rule] AWS IAM Backdoor User Key Creation
## Description

Detection of AWS API key creation for a user by another user. If there are Backdoored users in the environment, then the user can be used to obtain persistence in the AWS environment. 

### Target indexes

`filebeat-*, logs-aws*`

### Platforms

`AWS Cloudtrail`

### Query

```
event.dataset:aws.cloudtrail and event.provider:iam.amazonaws.com and event.action:CreateAccessKey and
(NOT (aws.cloudtrail.user_identity.arn.keyword:*responseElements.accessKey.userName*))
```

### **False Positives**

- Adding User Keys to their own Accounts

### MITRE

| Tactic      | Technique ID | Technique Name       | Sub-Technique Name |
| ----------- | ------------ | -------------------- | ------------------ |
| Persistence | T1098        | Account Manipulation |                    |

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1128 

## Summary

> Detection of AWS API key creation for a user by another user. If there are Backdoored users in the environment,  then the user can be used to obtain persistence in the AWS environment.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/aws/persistence_iam_backdoor_user_key_creation.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+risk_score = 21
+rule_id = "758dca01-0a25-408f-810f-7244eab892e2"
+severity = "low"
+tags = ["Elastic", "Cloud", "AWS", "Continuous Monitoring", "SecOps", "Identity and Access"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset:aws.cloudtrail and event.provider:iam.amazonaws.com and 
+event.action:CreateAccessKey and (not (aws.cloudtrail.user_identity.arn.keyword:*responseElements.accessKey.userName*))
```
Comment (bm11100) [inline rules/aws/persistence_iam_backdoor_user_key_creation.toml:36]: Do you have example data of this alert triggering?

File: rules/aws/persistence_iam_backdoor_user_key_creation.toml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+[metadata]
+creation_date = "2021/04/22"
+maturity = "production"
+updated_date = "2021/04/22"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detection of AWS API key creation for a user by another user. If there are Backdoored users in the environment, 
+then the user can be used to obtain persistence in the AWS environment.
```
Comment (brokensound77) [inline rules/aws/persistence_iam_backdoor_user_key_creation.toml:10]:
```suggestion
Detection of AWS API key creation for a user by another user. If there are backdoored user accounts in the environment, 
then those accounts can be used to obtain persistence in the AWS environment.
```

File: rules/aws/persistence_iam_backdoor_user_key_creation.toml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+"""
+false_positives = [
```
Comment (brokensound77) [inline rules/aws/persistence_iam_backdoor_user_key_creation.toml:12]: update or remove

File: rules/aws/persistence_iam_backdoor_user_key_creation.toml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+language = "kuery"
+license = "Elastic License v2"
+name = "AWS IAM Backdoor User Key Creation"
+note = """## Config
+
+The AWS Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
+
+references = [
+    "https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateAccessKey.html",
+]
```
Comment (brokensound77) [inline rules/aws/persistence_iam_backdoor_user_key_creation.toml:29]:
```suggestion
references = ["https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateAccessKey.html"]
```

File: rules/aws/persistence_iam_backdoor_user_key_creation.toml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+severity = "low"
+tags = ["Elastic", "Cloud", "AWS", "Continuous Monitoring", "SecOps", "Identity and Access"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset:aws.cloudtrail and event.provider:iam.amazonaws.com and 
+    event.action:CreateAccessKey and 
+                 not aws.cloudtrail.user_identity.arn.keyword:"*responseElements.accessKey.userName*"
+event.outcome:success
```
Comment (brokensound77) [inline rules/aws/persistence_iam_backdoor_user_key_creation.toml:41]:
```suggestion
event.dataset:aws.cloudtrail and event.provider:iam.amazonaws.com and 
    event.action:CreateAccessKey and 
    not aws.cloudtrail.user_identity.arn.keyword:"*responseElements.accessKey.userName*" and
    event.outcome:success    
```

File: rules/aws/persistence_iam_backdoor_user_key_creation.toml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+                 not aws.cloudtrail.user_identity.arn.keyword:"*responseElements.accessKey.userName*"
+event.outcome:success    
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1098"
+name = "Account Manipulation	"
```
Comment (brokensound77) [inline rules/aws/persistence_iam_backdoor_user_key_creation.toml:49]:
```suggestion
name = "Account Manipulation"
```

File: rules/aws/persistence_iam_backdoor_user_key_creation.toml (hunk 1 above)
Comment (brokensound77) [reply rules/aws/persistence_iam_backdoor_user_key_creation.toml]: Any update?

Comment (w0rk3r) [reply rules/aws/persistence_iam_backdoor_user_key_creation.toml]: @austinsonger any updates on this?

File: rules/aws/persistence_iam_backdoor_user_key_creation.toml (hunk 3 above)
Comment (w0rk3r) [reply rules/aws/persistence_iam_backdoor_user_key_creation.toml:12]: @austinsonger action needed

**Comment by w0rk3r** [comment]
Closing this one due to lack of activity, but feel free to reopen and update it

**Pull request closed without merging**